api/src/main/java/org/openmrs/module/bedmanagement/dao/BedManagementDao.java

```diff
@@ -50,4 +52,67 @@
     Bed getLatestBedByVisit(String visitUuid);
 
     List<BedTag> getAllBedTags();
+
+    /**
+     * Get all admission locations
+     *
+     * @return {@link List<AdmissionLocation>}
+     */
+    List<AdmissionLocation> getAdmissionLocations();
+
+    /**
+     * Get admission location by location
+     *
+     * @param location {@link Location}
+     * @return {@link AdmissionLocation} return null if not exist
+     */
+    AdmissionLocation getAdmissionLocationsByLocation(Location location);
```

Comment:
Please refactor into `getAdmissionLocationForLocation` (**no plural**, it returns a single object) and set this as the JavaDoc in both the service and DAO layer interfaces:
```
/**
  * Get the admission location for a given location.
  * Locations marked with the appropriate tag become admission locations. This method
  * returns the admission location, if applicable, for a given location.
  *
  * @param location {@link Location}
  * @return The {@link AdmissionLocation} or null if the location is not an admisson location.
  */
```